src/dotnet/commands/dotnet-install/dotnet-install-tool/InstallToolCommand.cs

```diff
@@ -82,55 +83,98 @@ public override int Execute()
+                    string.Format(
+                        LocalizableStrings.NuGetConfigurationFileDoesNotExist,
+                        Path.GetFullPath(_configFilePath)));
+            }
 
-                using (var transactionScope = new TransactionScope())
+            // Prevent installation if any version of the package is installed
+            if (_toolPackageStore.GetInstalledPackages(_packageId).FirstOrDefault() != null)
+            {
+                _errorReporter.WriteLine(string.Format(LocalizableStrings.ToolAlreadyInstalled, _packageId).Red());
+                return 1;
+            }
+
+            FilePath? configFile = null;
+            if (_configFilePath != null)
+            {
+                configFile = new FilePath(_configFilePath);
+            }
+
+            try
+            {
+                IToolPackage package = null;
+                using (var scope = new TransactionScope())
                 {
-                    var toolConfigurationAndExecutablePath = _toolPackageObtainer.ObtainAndReturnExecutablePath(
+                    package = _toolPackageInstaller.InstallPackage(
                         packageId: _packageId,
                         packageVersion: _packageVersion,
-                        nugetconfig: configFile,
-                        targetframework: _framework,
+                        targetFramework: _framework,
+                        nugetConfig: configFile,
                         source: _source,
                         verbosity: _verbosity);
 
-                    var commandName = toolConfigurationAndExecutablePath.Configuration.CommandName;
+                    foreach (var command in package.Commands)
+                    {
+                        _shellShimRepository.CreateShim(command.Executable, command.Name);
+                    }
 
-                    _shellShimMaker.CreateShim(
-                        toolConfigurationAndExecutablePath.Executable,
-                        commandName);
+                    scope.Complete();
+                }
 
-                    _environmentPathInstruction
-                        .PrintAddPathInstructionIfPathDoesNotExist();
+                _environmentPathInstruction.PrintAddPathInstructionIfPathDoesNotExist();
 
-                    _reporter.WriteLine(
-                        string.Format(LocalizableStrings.InstallationSucceeded, commandName));
-                    transactionScope.Complete();
-                }
+                _reporter.WriteLine(
+                    string.Format(
+                        LocalizableStrings.InstallationSucceeded,
+                        string.Join(", ", package.Commands.Select(c => c.Name)),
+                        package.PackageId,
+                        package.PackageVersion).Green());
+                return 0;
             }
-            catch (PackageObtainException ex)
+            catch (ToolPackageException ex)
             {
+                if (Reporter.IsVerbose)
```

Comment:
Why do you need to have this IsVerbose check? If you write to Verbose, it will only be outputted to the console if Verbose is turned on.

Reply:
The thinking was to avoid expense of `Exception.ToString` (my kingdom for compile-time macros ala-Rust 😭 ), but it's truly minimal and only in the error case anyway.  I'd be happy to remove the check.